### PR TITLE
Enhance handling of tags in FieldLayout

### PR DESCRIPTION
**Pull request by bartgol**
This PR should make it easier to add new field layouts without having to add new FieldTag enum's.

The main modifications are:
- Store tag names inside the layout. By default, we use `e2str(tag)` for each tag, but the user can overwrite with specific values.
- Remove Rad-related FieldTag values. Radiation will use `CMP`, setting the tag name to "swband"/"lwband" and the likes
- Minor change in some layout methods. The `strip_dim` method now modifies the current layout, so one has to do `fl.clone().strip_dim(CMP)` to obtain a copy of `fl` with the dimension stripped.
- Some fixes for handling tensor fields

### CODE REVIEW

File: components/eamxx/src/physics/mam/impl/compute_o3_column_density.cpp [hunk 1]
```diff
@@ -3,28 +3,27 @@ namespace scream::impl {
 KOKKOS_INLINE_FUNCTION
 void compute_o3_column_density(const ThreadTeam& team, const haero::Atmosphere& atm,
                                const mam4::Prognostics &progs, ColumnView o3_col_dens) {
-  constexpr int nabscol = mam4::gas_chemistry::nabscol;     // number of absorbing densities
   constexpr int gas_pcnst = mam4::gas_chemistry::gas_pcnst; // number of gas phase species
   constexpr int nfs = mam4::gas_chemistry::nfs;             // number of "fixed species"
-  constexpr Real mwdry = 1.0/haero::Constants::molec_weight_dry_air;
+  // constexpr Real mwdry = 1.0/haero::Constants::molec_weight_dry_air;
```
Comment (bartgol) [inline components/eamxx/src/physics/mam/impl/compute_o3_column_density.cpp:8]: All these commented lines are to get rid of compiler warnings due to unused stuff.

Comment (bartgol) [reply components/eamxx/src/physics/mam/impl/compute_o3_column_density.cpp:8]: Also, sometimes I removed the line, sometimes I commented, the rational being that
- if the  variable would be used in a function down below, but said function call is commented since it's not yet ported, then I commented the line declaring the var
- if the variable is _never_ used down below, then I removed the line altogether.

File: components/eamxx/tests/multi-process/dynamics_physics/homme_shoc_cld_p3_rrtmgp_pg2/CMakeLists.txt [hunk 2]
```diff
@@ -10,7 +10,7 @@ CreateDynamicsLib("theta-l_kokkos"  4   72   10)
 # Create the test
 CreateADUnitTest(${TEST_BASE_NAME}
   LABELS dynamics tms shoc cld p3 rrtmgp physics pg2
-  LIBS cld_fraction nudging tms shoc p3 scream_rrtmgp ${dynLibName} diagnostics
+  LIBS cld_fraction tms shoc p3 scream_rrtmgp ${dynLibName} diagnostics
```
Comment (tcclevenger) [inline components/eamxx/tests/multi-process/dynamics_physics/homme_shoc_cld_p3_rrtmgp_pg2/CMakeLists.txt:13]: What is the reason for this change?

**Review by tcclevenger — APPROVED** [review]
Just a question, but not blocking merging.

File: components/eamxx/tests/multi-process/dynamics_physics/homme_shoc_cld_p3_rrtmgp_pg2/CMakeLists.txt (hunk 2 above)
Comment (bartgol) [reply components/eamxx/tests/multi-process/dynamics_physics/homme_shoc_cld_p3_rrtmgp_pg2/CMakeLists.txt:13]: The test does not use nudging. So if you want to build just this test (while debugging), you would have also had to build nudging, for no reason.
